front/app/components/UI/QuillEditor/index.tsx

```diff
@@ -397,6 +397,25 @@ const QuillEditor = memo<Props & InjectedIntlProps>(
     const toolbarId = !noToolbar ? `ql-editor-toolbar-${id}` : null;
 
     const [editor, setEditor] = useState<Quill | null>(null);
+
+    useEffect(() => {
+      //a hacky way to add rtl to the underlying container of the quill editor.
+      // the "root" property is supposed to be INTERNAL according to the source code,
+      // but it works for now
+      if (editor) {
+        let root = editor.root;
```

Comment:
Is it possible to use `editor.format` instead? Here is a good suggestion for a similar implementation: https://github.com/quilljs/quill/issues/1632#issuecomment-360506379

It feels a bit cleaner.

Reply:
But this implementation is useful only if you want to make it dynamically changeable, actually. A much simpler approach would be to add this to the css whenever `theme.props.isRtl` is true or even use the `isRtl` helper directly from style utils:
```
.ql-editor {
  direction: rtl;
  text-align: right;
}
```

Is there perhaps a reason to have to add the styles in the React component that I am missing?

Reply:
> 

`editor.format()` didn't work for me for some reason.

Reply:
> 

I went with this approach instead, much cleaner.